### ISSUE TITLE
Scan detecting JBoss Fuse when should be JBoss AMQ

### ISSUE THREAD

**Issue by kdelee** [quipucords/rho#284]
## Specify type:
- Bug

### Bug severity (if applicable):
- High
___

## Description:
When scanning a RHEL5 system with JBoss A-MQ 6.2.0, rho reports that JBoss Fuse 6.2.0 is installed. (But there is no Fuse)
___

### Version of rho:
```
version: 0.0.28
	ansible: 2.3.2.0
	python: 3.5.3 (default, May 10 2017, 15:05:55)
```

### Expected behavior:
Currently the only field I see relating to "activemq" is `jboss.fuse.activemq-ver.
On a RHEL 6 machine, this produces:

jboss.fuse.activemq-ver | jboss.fuse.camel-ver | jboss.fuse.cxf-ver
------------------------------- | --------------------------- | -----------------------
 Unknown-Release: redhat-630262 |        |

I would expect the same from a RHEL 5 machine.
I do think that it is unclear there is no specific AMQ field.

### Actual behavior:
For a RHEL 5.9 server with JBoss A-MQ 6.2.0 installed, report actually contains:

jboss.fuse.activemq-ver | jboss.fuse.camel-ver | jboss.fuse.cxf-ver
------------------------------- | --------------------------- | -----------------------
Fuse-6.2.0 | Fuse-6.2.0 | Fuse-6.2.0

### Steps to reproduce:
On a RHEL 5.9 server, install OpenJDK 1.7 and  JBoss A-MQ 6.2.0 via zip file.
Scan the system with rho.

[Additional information about JBoss A-MQ 6.2.0 can be found in this guide](https://access.redhat.com/documentation/en-US/Red_Hat_JBoss_A-MQ/6.2/pdf/Installation_Guide/Red_Hat_JBoss_A-MQ-6.2-Installation_Guide-en-US.pdf)

### Environment information:
Provide detailed information on the *scanning* and *target* systems (if applicable); see example below.

Type            | Operating System | OS Version | Python Version | Virtualization
--------------- | ---------------- | ---------- | -------------- | --------------
Scanning System | Fedora             | 25        | 3.5            | Bare Metal
Target System   | RHEL             | 5.9        | 2.4.3          | ESXi

**Comment by chambridge** [comment]
@mdvickst Thoughts on the above bug?

@kdelee Only reason we didn't classify it as Fuse on RHEL 6 is because we don't have a classify mapping for the above **redhat-630262**.
https://github.com/quipucords/rho/blob/master/library/spit_results.py#L107

Looks like we need to understand how to differentiate between Fuse and AMQ better.

**Comment by mdvickst** [comment]
I'm not sure why this is happening. The logic that was given to me was that Fuse would have A-MQ, Camel and CXF where A-MQ would only have A-MQ jars. That is not the case for A-MQ 6.2.0 so we need to probably sync with the BU and get some better guidance.

**Comment by jdoyleoss** [comment]
There is a high level of reuse in the JBoss products. From a Product perspective, AMQ does not include Fuse, but it may have fuse jars as an internal implementation detail.

2nd, I would not expect any difference in scanning results from a different OS. If the same ZIP is producing different results on different OS versions, I suspect that it's something in the scanning tool, or perhaps the jars are from some other location in the filesystem.

**Comment by kdelee** [comment]
This is appearing with a specific version of A-MQ (back when it had a dash
in the name.)

I will dig into this more and try figure out what specific versions are
affected.

**Comment by kdelee** [comment]
So, just to elaborate, the version of A-MQ the RHEL 5 machine is 6.2.0, which was chosen because that is the most recent that supports Java 1.7, whereas AMQ 7 was chosen for RHEL 6 and 7 because they have openjdk 1.8 more readily available.

See these documents about supported configurations:
[AMQ 7 Supported Configurations](https://access.redhat.com/articles/2791941)
[A-MQ Supported Confugurations](https://access.redhat.com/articles/310613)

I guess one takeaway from this is that different major version should probably be treated as being almost different products --> i.e. assume nothing is the same.

**Comment by dhirajsb** [comment]
JBoss Fuse JON plugin uses multiple indicators to decide what product it is. It looks for the following in order:

1. A JON process query which looks for system properties and a main class:
```
    process|basename|match=^java.*,arg|-Dkaraf.home|match=.*,arg|-Dkaraf.base|match=.*,arg|*|match=org\.apache\.karaf\.(main|shell\.wrapper)\.Main
```
2. Java system properties `karaf.home` and `karaf.base` from above query

3. A product specific start script:
```
    JBoss Fuse - bin/fuse(.bat)?$
    JBoss A-MQ - bin/amq(.bat)?$
```
4. A version file specific to product:
```
    JBoss Fuse - system/org/jboss/fuse/jboss-fuse/(6\.[0-9\.\-(redhat)]+)
    JBoss A-MQ - system/org/jboss/amq/mq-fabric/(6\.[0-9\.\-(redhat)]+)
    JBoss A-MQ later version - system/io/fabric8/mq/mq-fabric/(1\.[0-9\.\-(redhat)]+)
```
So, product specific system properties, script files, and file name patterns should help differentiate between products.

**Comment by noahl** [comment]
@kdelee is the host that caused this one of the test hosts? I don't see an A-MQ machine in https://docs.google.com/spreadsheets/d/1-rfg-71h_lBqBK0s9g3J-mfFR-tTxqiRS-hfDm4P_js/edit?pli=1#gid=1464233263 . I'd like to reproduce if possible.

**Comment by kdelee** [comment]
@noahl I'll message you

**Comment by noahl** [comment]
If I understand correctly, the problem is that these three facts work by looking for jars, but the jars don't correspond to products in the way we thought they did.

I will talk with @mdvickst , but I think we should consider disabling these facts in the short term. I don't think they add as much as they used to, now that we have locate-based scans for Fuse. And we can't use the information from them to definitively identify Fuse without a lot of extra work to look at the path the jars are in, see if it's an EAP or Karaf base directory, and then try to identify JBoss products in that base directory.

If at some point we want to do that extra work, we could re-enable these facts.

@chambridge , what do you think?

**Comment by chambridge** [comment]
@noahl I think that is fine for now.

Alternatively we could update the current find commands to look for the jars and files that are in your locate commands as fingerprints instead of what's there currently since these find commands are essentially there for cases when locate is not there or not setup.

**Comment by noahl** [comment]
I'm almost done with a PR that will a) add new find-based commands to find the files we use for detecting Fuse and b) rename the facts in this issue to `jboss.activemq-ver`, `jboss.camel-ver` and `jboss.cxf-ver`. I think they should not have `fuse` in their name because that would suggest that they indicate fuse, which they don't. Without the `fuse`, they're just three facts that are disabled by default that detect some software we don't otherwise detect, so I think we can leave them in.

**Comment by noahl** [comment]
This was closed by #519 .
